Bio/SeqIO/SffIO.py

```diff
@@ -1159,14 +1135,23 @@ def __init__(self, handle, index=True, xml=None):
         """Initialize an SFF writer object.
 
         Arguments:
-         - handle - Output handle, ideally in binary write mode.
+         - handle - Output handle, in binary write mode.
          - index - Boolean argument, should we try and write an index?
          - xml - Optional string argument, xml manifest to be recorded
            in the index block (see function ReadRocheXmlManifest for
            reading this data).
 
         """
-        _check_mode(handle)
+        try:
+            # This should fail...
+            handle.write("ERROR - text mode handle\n")
+            raise ValueError(
+                "SFF files must NOT be opened in text mode, binary required."
+            )
+        except TypeError:
+            # Good, we wanted a binary handle:
+            # e.g. TypeError: a bytes-like object is required, not 'str'
+            pass
```

Comment:
Actually here you could use `handle.write(b"")` to explicitly test if the handle was opened in binary mode.

Reply:
I didn't think of that - almost but not quite a NOP (no operation), that does seem more elegant.